cycler.py

```diff
@@ -113,10 +113,26 @@ def __init__(self, left, right=None, op=None):
         Do not use this directly, use `cycler` function instead.
         """
         self._keys = _process_keys(left, right)
-        self._left = copy.deepcopy(left)
-        self._right = copy.deepcopy(right)
+        if isinstance(left, Cycler):
+            self._left = left._shallow_copy()
+        else:
+            self._left = copy.copy(left)
```

Comment:
This isn't deep enough to protect against changing the key, but this depends on how #13 ends up being implemented

```
In [80]: dd = [{a: a} for a in range(3)]

In [81]: import copy

In [82]: dd2 = copy.copy(dd)

In [83]: dd[0][0] = 'foobare'

In [84]: dd2
Out[84]: [{0: 'foobare'}, {1: 1}, {2: 2}]
```

I think this needs to be 

```python
self._left = [dict(v) for v in left]